lib/src/container/mod.rs

```diff
@@ -376,16 +377,34 @@ impl<'a> ManifestDiff<'a> {
 
 impl<'a> ManifestDiff<'a> {
     /// Prints the total, removed and added content between two OCI images
-    pub fn print(&self) {
-        let print_total = self.total;
-        let print_total_size = glib::format_size(self.total_size);
-        let print_n_removed = self.n_removed;
-        let print_removed_size = glib::format_size(self.removed_size);
-        let print_n_added = self.n_added;
-        let print_added_size = glib::format_size(self.added_size);
-        println!("Total new layers: {print_total:<4}  Size: {print_total_size}");
-        println!("Removed layers:   {print_n_removed:<4}  Size: {print_removed_size}");
-        println!("Added layers:     {print_n_added:<4}  Size: {print_added_size}");
+    pub fn print(&self, json: bool) {
+        if json {
+            let out = json!({
```

Comment:
Instead of this we can add `#[derive(Serialize)]` for ManifestDiff and get these bits for free.  We'll likely need to skip serializing the held references.